src/backend/joanie/payment/backends/lyra/__init__.py

```diff
@@ -480,3 +508,32 @@ def abort_payment(self, payment_id):
         """
         Abort a payment, nothing to do for Lyra
         """
+
+    def cancel_or_refund(self, amount: Money, transaction_reference: str):
+        """
+        Cancels or refunds a transaction made on the order's payment schedule.
+        The payment provider determines whether the transaction can be canceled or
+        refunded on their side. If the transaction has not yet been captured at the customer's
+        bank, it can be canceled. However, if the transaction has already been captured,
+        it means the funds have been received, and a refund process can be initiated instead.
+
+        https://docs.lyra.com/fr/rest/V4.0/api/playground/Transaction/CancelOrRefund
+        """
+        url = f"{self.api_url}Transaction/CancelOrRefund"
+
+        payload = {
+            "amount": int(amount.sub_units),
+            "currency": settings.DEFAULT_CURRENCY,
+            "uuid": str(transaction_reference),
+            "resolutionMode": "AUTO",
+        }
+
+        response_json = self._call_api(url, payload)
+
+        if response_json.get("status") != "SUCCESS":
+            raise exceptions.RegisterPaymentFailed(
+                f"The transaction reference {transaction_reference} does not "
+                "exist at the payment provider."
+            )
+
+        return True
```

Comment:
Same here, why returning a boolean ?

Reply:
I will return the `response_json` instead.